test/distributed/test_ddp_with_rpc.py

```diff
@@ -0,0 +1,444 @@
+        self,
+        remote_em_rref: rpc.RRef,
+        remote_net_rref: rpc.RRef,
+        process_group_for_ddp: dist.ProcessGroup = None,
+    ):
+        super(HybridModel, self).__init__()
+        self.remote_em_rref = remote_em_rref
+        self.remote_net_rref = remote_net_rref
+        self.remote_parameters_rrefs = _remote_method(
+            RemoteEM.get_parameter_rrefs, self.remote_em_rref
+        ) + _remote_method(RemoteNet.get_parameter_rrefs, self.remote_net_rref)
+        self.fc1 = nn.Linear(D_DENSE, D_DENSE)
+        self.fc2 = nn.Linear(D_HID, D_OUT)
+
+        if process_group_for_ddp is not None:
+            gLogger.info(f"Use DDP for the local nets.")
+            self.fc1 = DistributedDataParallel(
+                self.fc1,
+                process_group=process_group_for_ddp,
+                check_reduction=True,
+            )
+            self.fc2 = DistributedDataParallel(
+                self.fc2,
+                process_group=process_group_for_ddp,
+                check_reduction=True,
+            )
+
+        gLogger.info(
+            f"HybridModel has {len(list(self.parameters()))} groups of parameters."
+        )
+
+    def forward(self, input: FeatureSet):
+        gLogger.debug(f"Running HybridModel.forward on {input}")
+        em_features = _remote_method(
+            RemoteEM.forward, self.remote_em_rref, input.sparse_features
+        )
+        # The same size of mini batch.
+        assert em_features.shape[0] == input.dense_features.shape[0]
+        dense = self.fc1(input.dense_features)
+        x = torch.cat((dense, em_features), 1)
+        gLogger.debug(f"Concatenated feature: {x}")
+        x = _remote_method(RemoteNet.forward, self.remote_net_rref, x)
+        return self.fc2(x)
+
+    def print_parameters(self):
+        gLogger.debug(f"Local parameters: {list(self.parameters())}")
+        gLogger.debug(
+            f"Remote parameters: {[param_rref.to_here() for param_rref in self.remote_parameters_rrefs]}"
+        )
+
+
+class Trainer:
+    def __init__(
+        self,
+        remote_em_rref: rpc.RRef,
+        remote_net_rref: rpc.RRef,
+        ddp_mode: DdpMode,
+        rank: int,
+    ):
+        gLogger.info(
+            f"Initializing trainer process group by traner #{rank} with ranks {TRAINER_RANKS}"
+        )
+        self.process_group_for_ddp = dist_c10d.new_group(ranks=TRAINER_RANKS)
+        check_process_group_type(f"trainer #{rank}", self.process_group_for_ddp)
+
+        self.criterion = torch.nn.CrossEntropyLoss()
+        self.remote_em_rref = remote_em_rref
+        self.remote_net_rref = remote_net_rref
+        self.hybrid_module = HybridModel(
+            self.remote_em_rref,
+            self.remote_net_rref,
+            self.process_group_for_ddp
+            if ddp_mode in (DdpMode.INSIDE,)
+            else None,
+        )
+        if ddp_mode == DdpMode.OUTSIDE:
+            gLogger.info(f"Wrapping the whole hybride module into DDP.")
```

Comment:
I think you meant to write *hybrid* instead of *hybride*.

```suggestion
            gLogger.info(f"Wrapping the whole hybrid module into DDP.")
```